packages/pg-pubsub/src/PgSubscriptionResolverPlugin.ts

```diff
@@ -101,13 +102,41 @@ const PgSubscriptionResolverPlugin: Plugin = function(builder, { pubsub }) {
                 "filter provided to pgSubscription must be a function"
               );
             }
-            return withFilter(() => asyncIterator, filter)(
+            asyncIterator = withFilter(() => asyncIterator, filter)(
               parent,
               args,
               resolveContext,
               resolveInfo
             );
           }
+
+          if (initialEvent) {
+            if (typeof initialEvent !== "function") {
+              throw new Error(
+                "initialEvent provided to pgSubscription must be a function"
+              );
+            }
+
+            return (async function* subscribeWithInitialEvent() {
+              const event = await initialEvent(
+                args,
+                resolveContext,
+                resolveInfo
+              );
+              if (event === null || typeof event !== "object") {
+                throw new Error(
+                  "initialEvent returning event must be an object"
+                );
+              }
+              yield { ...event, topic };
+              for await (const val of {
+                [Symbol.asyncIterator]: () => asyncIterator,
+              }) {
```

Comment:
Is this really required? I thought that the following would work?
```suggestion
              for await (const val of asyncIterator) {
```

Reply:
It is valid, but TS complains. Instead of doing a `// @ts-ignore`, I just complied.

Reply:
Hmmm... We should appease TypeScript rather than using this non-obvious syntax.